pyccel/codegen/codegen.py

```diff
@@ -9,31 +9,39 @@
 from pyccel.codegen.printing.fcode  import FCodePrinter
 from pyccel.codegen.printing.ccode  import CCodePrinter
 from pyccel.codegen.printing.pycode import PythonCodePrinter
+from pyccel.codegen.printing.cucode import CudaCodePrinter
 
 from pyccel.ast.core      import FunctionDef, Interface, ModuleHeader
 from pyccel.errors.errors import Errors
 from pyccel.utilities.stage import PyccelStage
 
-_extension_registry = {'fortran': 'f90', 'c':'c',  'python':'py'}
-_header_extension_registry = {'fortran': None, 'c':'h',  'python':None}
+_extension_registry = {'fortran': 'f90', 'c':'c',  'python':'py', 'cuda':'cu'}
+_header_extension_registry = {'fortran': None, 'c':'h',  'python':None, 'cuda':'h'}
 printer_registry    = {
                         'fortran':FCodePrinter,
                         'c':CCodePrinter,
-                        'python':PythonCodePrinter
+                        'python':PythonCodePrinter,
+                        'cuda':CudaCodePrinter
                       }
 
 pyccel_stage = PyccelStage()
 
 class Codegen(object):
 
-    """Abstract class for code generator."""
+    """
+    Abstract class for code generator.
+
+    Handles the collect of statements, initiating the correct printer and Generating the files.
+    """
 
     def __init__(self, parser, name):
-        """Constructor for Codegen.
+        """
+        Constructor for Codegen.
 
+        Parameters
+        ----------
         parser: pyccel parser
-
-
+            The Pyccel parser node.
         name: str
             name of the generated module or program.
         """
```

Comment:
Docstrings in `__init__` are more or less useful as they cannot be parsed. The class docstring should describe the `__init__` interface. See https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring